components/input/function-call-form.tsx

```diff
@@ -33,7 +34,14 @@ export const FunctionCallForm: FC<FunctionCallFormProps> = ({ onAddAction }) =>
           placeholder="0x1234..."
           variant={!targetContract || isAddress(targetContract) ? "default" : "critical"}
           value={targetContract}
-          onChange={(e) => setTargetContract(e.target.value || "")}
+          onChange={(e) => {
+            try {
+              const address = getAddress(e.target.value);
+              setTargetContract(address);
+            } catch (e) {
+              setTargetContract(undefined);
```

Comment:
This will work if you copy the address. But if for some reason you entered it manually, it would always reset the value to undefined.

We could show an error alert instead and simply assume that you will always copy addresses. But I would never change the user's value without proper feedback about what happened

Reply:
You are right, I rolled back the `targetContract` type to `string` and now I use `as Address` where is needed. But I noticed that we are using our own `isAddress` function instead of the provided by Viem. Is there any reason for this? I found out the Viem checker also checks for checksums.

Reply:
I wasn't aware of viem's version, so it would be ideal if we switch into using it

Reply:
Np, I'll open a ticket for it 🎫